### PR TITLE
Get lustre, zfs and robinhood from external repos

### PR DESCRIPTION
**Pull request by brianjmurrell**
For now we will just use the Jenkins job lustre-master and the yum
repo that is created in it's artefacts.  When a final Lustre release
is made this should switch to the official download location.

### CODE REVIEW

**Comment by tanabarr** [comment]
this seems to introduce a sim test failure

**Comment by brianjmurrell** [comment]
@tanabarr: Yes.  This patch is very much a work in progress still.  It's going to touch everything to do with bundle (or lack thereof) and package installation, etc.  It is expected that it's going to fail lots of tests until it's complete.

This isn't a patch that anyone should be basing PRs on at this point.

File: cluster-sim/cluster_sim/simulator.py [hunk 1]
```diff
@@ -243,10 +243,12 @@ def setup(self, server_count, worker_count, volume_count, nid_count, cluster_siz
...
             'server': {
                 'lustre': (0, '2.1.4', '1', 'x86_64'),
-                'lustre-modules': (0, '2.1.4', '1', 'x86_64'),
-                'lustre-osd-ldiskfs': (0, '2.1.4', '1', 'x86_64'),
-                'lustre-osd-zfs': (0, '2.1.4', '1', 'x86_64'),
-                'kernel-devel-3.10.0-514.10.2.el7_lustre': (0, '2.6.32', '1', 'x86_64'),
+                'lustre-dkms': (0, '2.1.4', '1', 'x86_64'),
+                'kmod-lustre-osd-ldiskfs': (0, '2.1.4', '1', 'x86_64'),
+                'lustre-osd-zfs-mount': (0, '2.1.4', '1', 'x86_64'),
+                'kernel-[0-9]*_lustre': (0, '2.6.32', '1', 'x86_64'),
```
Comment (tanabarr) [inline cluster-sim/cluster_sim/simulator.py]: are you sure that glob is expanding as you expect it to here and not causing silent error?

Comment (brianjmurrell) [inline cluster-sim/cluster_sim/simulator.py]: The glob is not expected to expand in the simulator.  It's just meant to match what is in the profile.

Not saying it's not a source of trouble, but if it is, that's a simulator problem, not a profile problem.

**Comment by tanabarr** [comment]
unit tests fixed with updates to audit and metrics lustre plug-in tests and code, sample data updated
EFS tests fixed with changes to install-client.sh to ensure we are installing the kmod-lustre-client package onto the kernel it was built for
sim & SSI test fixes in progress...

File: chroma-manager/tests/integration/shared_storage_configuration/test_updates.py [hunk 2]
```diff
@@ -28,18 +27,21 @@ def test_upgrade_alerting(self):
 
         self.assertNotEqual(len(original_packages), 0)
 
-        self.assertNoAlerts(host['resource_uri'], of_type='UpdatesAvailableAlert')
+        if config.get('simulator'):
+            # the simulator only does a package scan on initial (fake) plug-in update,
+            # therefore we have to do it manually to avoid race conditions failing this test
```
Comment (brianjmurrell) [inline chroma-manager/tests/integration/shared_storage_configuration/test_updates.py]: Is this a race we just **always** won before this patch or is there some other difference that is requiring that we force this `scan_packages()` now?

Comment (tanabarr) [inline chroma-manager/tests/integration/shared_storage_configuration/test_updates.py]: it is hard to tell with the number of different changes in this patch

**Comment by tanabarr** [comment]
replaced by PR 119

**Comment by brianjmurrell** [comment]
This is not replaced by #119.  #119 was just used to work out the right solution.

**Comment by brianjmurrell** [comment]
@tanabarr Somewhat agreed.  Unfortunately the commit history was a large haystack of debugging mess anyway with a low signal to noise ratio.

I think in the future, my process to mitigate that problem is going to be to do exploratory debug work in a separate PR/branch based off of the PR/branch I'm debugging and then squashing the results of the debug PR/branch into a reasonable number of useful commits (probably most likely just one) and then pushing (without rebase) that one commit to the PR/branch that the debugging was being done for (i.e. the one the deubgging PR/branch was based on).

This I think should keep the debugging noise in the main PR/branch to a minimum (if not eliminate it completely) and keep the commit history useful.

File: chroma-agent/tests/actions_plugins/test_agent_updates.py [hunk 3]
```diff
@@ -67,32 +68,25 @@ def test_unconfigure_repo_no_file(self):
...
-                return "2.6.32-358.2.1.el6.x86_64\n"
-            elif args == ["rpm", "-q", "kernel"]:
-                return """kernel-2.6.32-358.2.1.el6.x86_64
-kernel-2.6.32-358.18.1.el6.x86_64
-"""
-
-        with patch('chroma_agent.lib.shell.AgentShell.try_run', side_effect=try_run):
+        def run(arg_list):
+            values = {("rpm", "-q", "--whatprovides", "kmod-lustre"):
+                      "kmod-lustre-1.2.3-1.el6.x86_64\n",
```
Comment (tanabarr) [inline chroma-agent/tests/actions_plugins/test_agent_updates.py]: we should try not to be mentioning el6 in tests

File: chroma-manager/Makefile [hunk 4]
```diff
@@ -149,7 +149,7 @@ agent_tests:
 			| tee ../chroma-agent/unit.log; \
 	exit $${PIPESTATUS[0]}
 
-unit_tests:
+unit_tests unit-tests:
```
Comment (tanabarr) [inline chroma-manager/Makefile]: out of curiosity , why is this required?

**Comment by brianjmurrell** [comment]
> all the changes I have made on this patch are now listed as being made by you, please can you fix to preserve accurate history

What would you suggest I do?  We could add a signed-off on the commit message.

File: chroma-agent/chroma_agent/action_plugins/agent_updates.py [hunk 5]
```diff
@@ -161,20 +160,30 @@ def kernel_status():
...
-        except AgentShell.CommandExecutionError:
-            required_kernel_stdout = None
-
-    required_kernel = None
-    if required_kernel_stdout:
-        for line in required_kernel_stdout.split("\n"):
-            if line.startswith('kernel'):
-                required_kernel = "kernel-%s.%s" % (line.split(" = ")[1],
-                                                    platform.machine())
+            required_kernel_prefix = \
```
Comment (brianjmurrell) [inline chroma-agent/chroma_agent/action_plugins/agent_updates.py]: I'm not sure I follow.  It is a prefix.

File: chroma-agent/tests/actions_plugins/test_agent_updates.py (hunk 3 above)
Comment (brianjmurrell) [inline chroma-agent/tests/actions_plugins/test_agent_updates.py]: It doesn't really matter.  It's a string that is part of a package release.  It has no meaning whatsoever.  It could say "blat".

File: chroma-manager/Makefile (hunk 4 above)
Comment (brianjmurrell) [inline chroma-manager/Makefile]: So that one can say "make unit-tests" if that is what one is used to.  It's causing no harm.
